dhcpv6/dhcpv6relay.go

```diff
@@ -182,29 +182,27 @@ func (r *DHCPv6Relay) GetInnerPeerAddr() (net.IP, error) {
 	return addr, nil
 }
 
-// NewRelayReplFromRelayForw creates a RELAY_REPLY packet based on a RELAY_FORW
-// packet containing the passed DHCPv6 message as payload.
+// NewRelayReplFromRelayForw creates a RELAY_REPL packet based on a RELAY_FORW
+// packet and replaces the inner message with the passed DHCPv6 message.
 func NewRelayReplFromRelayForw(relayForw, msg DHCPv6) (DHCPv6, error) {
 	var (
 		err                error
 		linkAddr, peerAddr []net.IP
 		optiids            []Option
 	)
-
 	if relayForw == nil {
 		return nil, errors.New("RELAY_FORW cannot be nil")
 	}
 	if relayForw.Type() != RELAY_FORW {
-		return nil, errors.New("The passed RELAY_FORW must have RELAY_FORW type set")
+		return nil, errors.New("The passed packet is not of type RELAY_FORW")
 	}
+	relay := relayForw.(*DHCPv6Relay)
```

Comment:
what I actually meant is to do

```
relay, ok := relayForw.(*DHCPv6Relay)
if !ok {
    return nil, errors.New("Not a DHCPv6Relay")
}
if relay.Type() != RELAY_FORW {
    return nil, errors.New("The passed packet is not of type RELAY_FORW")
}

Reply:
ok, thanks for the clarification, I updated it